recipes/ffmpeg/all/conanfile.py

```diff
@@ -543,57 +537,57 @@ def opt_append_disable_if_set(args, what, v):
             ])
         if not self.options.with_programs:
             args.append("--disable-programs")
-        # since ffmpeg's build system ignores toolchain variables
-        if tools.get_env("AR"):
-            args.append("--ar={}".format(tools.get_env("AR")))
-        if tools.get_env("AS") and self.options.with_asm:
-            args.append("--as={}".format(tools.get_env("AS")))
-        if tools.get_env("CC"):
-            args.append("--cc={}".format(tools.get_env("CC")))
-        if tools.get_env("CXX"):
-            args.append("--cxx={}".format(tools.get_env("CXX")))
-        if tools.get_env("NM"):
-            args.append("--nm={}".format(tools.get_env("NM")))
-        if tools.get_env("RANLIB"):
-            args.append("--ranlib={}".format(tools.get_env("RANLIB")))
-        if tools.get_env("STRIP"):
-            args.append("--strip={}".format(tools.get_env("STRIP")))
-        extra_cflags = []
-        extra_ldflags = []
-        if is_apple_os(self) and self.settings.os.version:
-            extra_cflags.append(tools.apple_deployment_target_flag(
-                self.settings.os, self.settings.os.version))
-            extra_ldflags.append(tools.apple_deployment_target_flag(
-                self.settings.os, self.settings.os.version))
+        # since ffmpeg"s build system ignores CC and CXX
+        compilers_from_conf = self.conf.get("tools.build:compiler_executables", default={}, check_type=dict)
+        buildenv_vars = VirtualBuildEnv(self).vars()
+        asm = compilers_from_conf.get("asm", buildenv_vars.get("AS"))
+        if asm:
+            args.append(f"--as={unix_path(self, asm)}")
+        cc = compilers_from_conf.get("c", buildenv_vars.get("CC"))
+        if cc:
+            args.append(f"--cc={unix_path(self, cc)}")
+        cxx = compilers_from_conf.get("cpp", buildenv_vars.get("CC"))
+        if cxx:
+            args.append(f"--cxx={unix_path(self, cxx)}")
+        pkg_config = self.conf.get("tools.gnu:pkg_config", default=buildenv_vars.get("PKG_CONFIG"), check_type=str)
+        if pkg_config:
```

Comment:
I think the flags for ar, nm, ranlib and strip are missing. This might cause some issues while cross compiling. At least it was an issue with conan1

Reply:
fixed by https://github.com/conan-io/conan-center-index/pull/15819/commits/5f0359f0a26fb43f1e4c52b407be65e4e7f8dd68